### PR TITLE
EHPR-275 - form updates

### PR DESCRIPTION
**Pull request by coryatwater-FW**
### Pending
- confirmation of which HA's to show in previous deployment dropdown

### Form Updates 1
- Split “Emergency Responder” stream type into (i) Paramedic and (ii) Firefighter. Subspecialities for Paramedic to include (i) EMR, (ii) PCP, (iii) ACP, (iv) Critical Care (CCP). Subspeciality for Firefighter to include only FR (Fire First responder).
- add conditional fields for `Have you previously been deployed from the EHPR?` = `Yes`
- Date of Last Deployment and Last Deployed Health Authority
- update `Indicate the maximum duration of deployment you are willing to support` options
- update landing page text, added `You will be required to undergo a criminal record check or criminal record check verification under the Criminal Records Review Act prior to any deployment. Individuals who forgo the criminal record check or criminal record check verification process will not be eligible for deployment and may be removed from this registry.`

Screenshot 2024-02-21 at 9 05 02 AM

Screenshot 2024-02-21 at 9 04 40 AM

Screenshot 2024-02-22 at 2 11 02 PM

Screenshot 2024-02-21 at 9 04 24 AM

### Ticket
[EHPR-275](https://eydscanada.atlassian.net/jira/software/c/projects/EHPR/boards/260?selectedIssue=EHPR-275)

### CODE REVIEW

File: apps/web/src/components/submission/components/Preferences.tsx [hunk 1]
```diff
@@ -80,6 +94,37 @@ export const Preferences: React.FC<FormStepProps> = () => {
...
+              format='yyyy-MM-dd'
+              max={new Date()}
+            />
+          </div>
+          <div className='flex flex-row'>
+            <Field
+              name={'preferencesInformation.lastDeployedHa'}
+              label={'Last Deployed Health Authority'}
+              component={({ field, form }: FieldProps) => (
+                <ReactSelect<OptionType>
```
Comment (dbayly-freshworks) [inline apps/web/src/components/submission/components/Preferences.tsx:112]: Looks like this is getting flagged by sonar cloud

**Review by dbayly-freshworks — COMMENTED** [review]
One comment, otherwise this looks good.

File: apps/web/src/components/submission/components/Preferences.tsx (hunk 1 above)
Comment (coryatwater-FW) [reply apps/web/src/components/submission/components/Preferences.tsx]: seems to be an issue with the select being nested, I looked through the other issues in sonarcloud and it seems to flag all occurrences of this ReactSelect, I shall look into it

File: apps/web/src/components/submission/components/Credential.tsx [hunk 2]
```diff
@@ -129,7 +127,7 @@ export const Credential: React.FC = () => {
                   <legend className='font-semibold mb-4'>Select your specialties</legend>
                   {specialties?.map((_, index) => (
                     <SpecialtySelector
-                      key={index}
+                      key={_.id}
```
Comment (dbayly-freshworks) [inline apps/web/src/components/submission/components/Credential.tsx:130]: Nit - Might be good to have a variable name here for readability.

**Review by dbayly-freshworks — APPROVED** [review]
Looks good - one nit

**Pull request merged**